addons/website_sale/models/delivery_carrier.py

```diff
@@ -12,3 +16,53 @@ class DeliveryCarrier(models.Model):
         related='product_id.description_sale',
         readonly=False,
     )
+    has_estimated_delivery = fields.Boolean(
+        string='Estimated Delivery',
+        help="Display estimated date to your customer. Editable if range is defined."
+    )
+    estimated_delivery_start = fields.Integer()
+    estimated_delivery_end = fields.Integer()
+    opening_hours = fields.Many2one(
+        string="Opening Hours", comodel_name='resource.calendar', check_company=True
+    )
+
+    def _get_available_days(self):
+        """Return the available days defined on the estimated delivery field based on the calendar.
+
+        :returns: list of all available days of a given range.
+        :rtype: list
+        """
+        self.ensure_one()
+        available_days = []
+        availability_range = self.estimated_delivery_end + self.estimated_delivery_start
+        if self.has_estimated_delivery:
+            # `_attendance_intervals_batch` requires the datetime to be timezoned
+            tz = timezone(self.opening_hours.tz)
+            current_date = fields.Datetime.now()
+            from_datetime = (current_date).astimezone(tz)
+            # add 30 days as buffer to account for unavailable days
+            to_datetime = (
+                current_date + timedelta(days=self.estimated_delivery_end + 30)
+            ).astimezone(tz)
+            availabilities = self.opening_hours._attendance_intervals_batch(
+                from_datetime, to_datetime)[False]._items
+            for availability in availabilities:
```

Comment:
Why do you access `_items`? You can iterate over the `Intervals` object.
```suggestion
            availabilities = self.opening_hours._attendance_intervals_batch(
                from_datetime, to_datetime
            )[False]
            for availability in availabilities:
```